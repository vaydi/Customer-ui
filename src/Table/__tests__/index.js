import itRendersAllMutations from 'jest-it-renders-all-mutations';

import TableDefault from '..';

const mutations = [
  {
    name: 'with all data',
    props: {
      labels: ['1', '2', '3'],
      data: [{ 1: '1' }, { 2: '2' }, { 3: '3' }],
      extraDataFormating: [],
      pending: false,
      noDataMessage: 'Aucune donnÃ©e',
      pagination: false,
      rowsPerPage: 10,
    },
  },
  {
    name: 'without data',
    props: {
      labels: [],
      data: [],
    },
  },
];

describe(TableDefault.name, () => {
  itRendersAllMutations(TableDefault, mutations);
});
