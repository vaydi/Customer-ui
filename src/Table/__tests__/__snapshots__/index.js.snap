// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithStyles with all data renders correctly 1`] = `
<div>
  <WithStyles(Table)
    className="TableDefault-table-1"
  >
    <WithStyles(TableHead)>
      <WithStyles(TableRow)>
        <WithStyles(TableCell)
          className="TableDefault-head-4"
          key="1"
        >
          1
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          className="TableDefault-head-4"
          key="2"
        >
          2
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          className="TableDefault-head-4"
          key="3"
        >
          3
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
    </WithStyles(TableHead)>
    <WithStyles(TableBody)>
      <WithStyles(TableRow)
        className="TableDefault-row-2"
        key="0"
      >
        <WithStyles(TableCell)
          className="TableDefault-cell-3"
          key="00"
        >
          1
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
      <WithStyles(TableRow)
        className="TableDefault-row-2"
        key="1"
      >
        <WithStyles(TableCell)
          className="TableDefault-cell-3"
          key="10"
        >
          2
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
      <WithStyles(TableRow)
        className="TableDefault-row-2"
        key="2"
      >
        <WithStyles(TableCell)
          className="TableDefault-cell-3"
          key="20"
        >
          3
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
    </WithStyles(TableBody)>
  </WithStyles(Table)>
</div>
`;

exports[`WithStyles without data renders correctly 1`] = `
<div>
  <WithStyles(Table)
    className="TableDefault-table-1"
  >
    <WithStyles(TableHead)>
      <WithStyles(TableRow) />
    </WithStyles(TableHead)>
    <WithStyles(TableBody)>
      <WithStyles(TableRow)>
        <WithStyles(TableCell)
          colSpan={0}
        >
          Aucune donn√©e
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
    </WithStyles(TableBody)>
  </WithStyles(Table)>
</div>
`;
